app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -104,7 +105,9 @@ class KiwixMainActivity : CoreMainActivity() {
     activityKiwixMainBinding = ActivityKiwixMainBinding.inflate(layoutInflater)
     setContentView(activityKiwixMainBinding.root)
     if (intent.action == "GET_CONTENT") {
-      navigate(R.id.downloadsFragment)
```

Comment:
@shashank-0-0 The back stack is not broken, if I click on the back button it opens the `reader` screen from the back stack so the issue is something else. We should fix the actual issue.


https://github.com/user-attachments/assets/c8f5d277-8090-4126-8fed-200ca0738dd7

Reply:
Looking in to it !!

Reply:
looking in to it !!

Reply:
@shashank-0-0 Any update here?

Reply:
@MohitMaliFtechiz my bad couldn't take the time ,
 i think this is precisely the issue here https://stackoverflow.com/a/71566840/15331061 . Our bottom nav items each have their own backstacks and when we navigateTo(a fragment) we just add the fragment to the respective back stack.
In our case we are  adding the download fragment to the Reader fragment's back stack so when we click on Reader tab we are taken to the top of its backstack which has Download fragment.
So i guess the fix is fine , can u look into it again thanks.

Reply:
@shashank-0-0 Thank you for your explanation i will have a look at this, in the meantime, can you please implement my other comment?